xds/src/main/java/io/grpc/xds/ServerXdsClient.java

```diff
@@ -76,7 +76,7 @@
     super(channel, useProtocolV3, node, timeService, backoffPolicyProvider, stopwatchSupplier);
     this.useNewApiForListenerQuery = useProtocolV3 && useNewApiForListenerQuery;
     this.instanceIp = (instanceIp != null ? instanceIp : "0.0.0.0");
-    this.grpcServerResourceId = grpcServerResourceId != null ? grpcServerResourceId : "grpc/server";
+    this.grpcServerResourceId = grpcServerResourceId;
```

Comment:
Does this now need a checkNotNull?

What error message is produced if this field isn't set? Does it just cause a NPE? Does there need to be a check earlier in the process to produce a clear error message? It looks like `XdsClientWrapperForServerSds.createXdsClientAndStart()` would be an appropriate place for the check, or by `BootstrapInfo.getGrpcServerResourceId()` itself.

Reply:
I did think about it and rejecting an unset/null value but then decided against. I think what'll happen is this client will just send a query for "null?udpa.listener.address=0.0.0.0:port" and then we'll get an error from TD. We should have a common implementation among languages so this might not be a good idea. I will add a check...

Reply:
added a check in `XdsClientWrapperForServerSds.createXdsClientAndStart()`